developers/articles/project-templates.md

```diff
@@ -0,0 +1,373 @@
+
+Project templates in the New Project dialog have three category levels. Xamarin Studio and MonoDevelop define the following category levels.
+
+- crossplat
+     - app
+         - forms
+         - native
+         - games
+     - library
+         - general
+     - tests
+         - general
+- ios
+    - app
+         - general
+         - games
+         - watch
+         - classic
+    - extension
+         - general
+         - watch
+    - library
+         - unified
+         - classic
+    - tests
+         - unified
+         - classic
+- android
+    - app
+         - general
+    - library
+         - general
+    - tests
+         - general
+- mac
+    - app
+         - general
+         - games
+         - classic
+    - library
+         - unified
+         - classic
+- other
+    - net
+         - general
+    - aspnet
+         - general
+    - misc
+         - generic
+         - c
+         - general
+    - unsupported
+         - monomac
+
+MonoDevelop’s categories are defined in the Templates.addin.xml file.
+
+Xamarin Studio’s categories are defined in the Xamarin.Ide.addin.xml file.
+
+### Adding a Project Template to an Existing Category
+
+To add your project template into one of the above categories all three category ids, one for each level, must be defined in the project template’s TemplateConfiguration section.
+
+``` xml
+<TemplateConfiguration>
+	<_Name>Console Project</_Name>
+	<Category>other/net/general</Category>
+       ...
+</TemplateConfiguration>
+```
+
+If MonoDevelop cannot match the project template’s category path to any existing categories then it will be put in the other/misc/general category by default.
+
+### Adding New Project Template Categories
+
+You are free to add your own third level categories. To add your own project template categories you can add them to your addin’s .addin.xml file:
+
+``` xml
+<Extension path="/MonoDevelop/Ide/ProjectTemplateCategories">
+	<Category id="toplevel" name="Top Level" icon="md-platform-other">
+		<Category id="second-level-a" name="Second Level A">
+			<Category id="third-level-a" name="Third Level A" />
+		</Category>
+		<Category id="second-level-b" name="Second Level B">
+			<Category id="third-level-b" name="Third Level B" />
+		</Category>
+	</Category>
+</Extension>
+```
+
+Only the top level category has an icon. These top level category icons are defined as resources in the .addin.xml file.
+
+``` xml
+<Extension path="/MonoDevelop/Core/StockIcons">
+  <StockIcon stockid="md-platform-other" resource="platform-other-light-16.png" size="Menu" />
+</Extension>
+```
+
+The icons should be compiled as embedded resources in your addin.
+
+MonoDevelop supports retina icons everywhere and to use them in the New Project dialog on a retina device just add a second embedded resource with a filename that @2x just before the file extension.
+
+- Non-retina resource: platform-other-light-16.png
+- Retina resource: platform-other-light-16@2x.png
```

Comment:
Please wrap this in backticks, otherwise it'll be rendered as a `mailto:` link due to the @.